src/main/scala/nl.knaw.dans.easy.multideposit/parser/FileDescriptorParser.scala

```diff
@@ -19,88 +19,60 @@ import better.files.File
+    val fileAccessibilities = dataPerPath.collect { case (_, _, Some(far), _) => far }
+    val fileVisibilities = dataPerPath.collect { case (_, _, _, Some(fv)) => fv }
+    val errors = ListBuffer[Throwable]()
 
-    (titles, fars) match {
-      case (Nil, Nil) => Success((file, FileDescriptor()))
-      case (t :: Nil, Nil) => Success((file, FileDescriptor(Some(t))))
-      case (Nil, f :: Nil) => Success((file, FileDescriptor(accessibility = Some(f))))
-      case (t :: Nil, f :: Nil) => Success((file, FileDescriptor(Some(t), Some(f))))
-      case (ts, fs) if fs.size <= 1 => Failure(ParseException(rowNum, s"FILE_TITLE defined multiple values for file '$file': ${ ts.mkString("[", ", ", "]") }"))
-      case (ts, fs) if ts.size <= 1 => Failure(ParseException(rowNum, s"FILE_ACCESSIBILITY defined multiple values for file '$file': ${ fs.mkString("[", ", ", "]") }"))
-      case (ts, fs) => Failure(new CompositeException(
-        ParseException(rowNum, s"FILE_TITLE defined multiple values for file '$file': ${ ts.mkString("[", ", ", "]") }"),
-        ParseException(rowNum, s"FILE_ACCESSIBILITY defined multiple values for file '$file': ${ fs.mkString("[", ", ", "]") }")))
+    if (titles.size > 1) errors.append(ParseException(rowNum, s"FILE_TITLE defined multiple values for file '$file': ${ titles.mkString("[", ", ", "]") }"))
+    if (fileAccessibilities.size > 1) errors.append(ParseException(rowNum, s"FILE_ACCESSIBILITY defined multiple values for file '$file': ${ fileAccessibilities.mkString("[", ", ", "]") }"))
+    if (fileVisibilities.size > 1) errors.append(ParseException(rowNum, s"FILE_VISIBILITY defined multiple values for file '$file': ${ fileVisibilities.mkString("[", ", ", "]") }"))
+
+    if (errors.nonEmpty) Failure(CompositeException(errors))
+    else {
+      (fileAccessibilities, fileVisibilities) match {
+        case (List(as), List(vs)) if vs > as => Failure(ParseException(rowNum,
+          s"FILE_VISIBILITY ($vs) is more restricted than FILE_ACCESSIBILITY ($as) for file '$file'. (User will potentially have access to an invisible file.)"))
+        case _ => Success((file, FileDescriptor(titles.headOption, fileAccessibilities.headOption, fileVisibilities.headOption)))
+      }
     }
   }
 
-  def fileDescriptor(depositId: DepositId)(rowNum: => Int)(row: DepositRow): Option[Try[(File, Option[String], Option[FileAccessRights.Value])]] = {
-    val path = row.find("FILE_PATH").map(findPath(depositId))
+  def fileDescriptor(depositId: DepositId)(rowNum: => Int)(row: DepositRow): Option[Try[(File, Option[String], Option[FileAccessRights.Value], Option[FileAccessRights.Value])]] = {
+    def collectErrors(rs: Option[Try[_]]*): Seq[Throwable] = {
+      rs.collect { case Some(Failure(t)) => ParseException(rowNum, t.getMessage, t) }
+    }
+
+    val path = row.find("FILE_PATH").map(findRegularFile(depositId))
     val title = row.find("FILE_TITLE")
     val accessRights = fileAccessRight(rowNum)(row)
```

Comment:
should this be renamed to 'accessibility' and 'fileAccessibility' to be more like the 'visibility' part?